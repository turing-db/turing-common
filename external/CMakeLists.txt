set(RANGE_V3_DOCS OFF)
set(RANGE_V3_TESTS OFF)
set(RANGE_V3_EXAMPLES OFF)
set(RANGE_V3_PERF OFF)

add_subdirectory(nlohmann_json)
add_subdirectory(crow)
add_subdirectory(range-v3)
add_subdirectory(argparse)
add_subdirectory(spdlog)
add_subdirectory(expected)
add_subdirectory(linenoise)
add_subdirectory(tabulate)
add_subdirectory(termcolor)
