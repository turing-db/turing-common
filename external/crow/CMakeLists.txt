
add_library(crow INTERFACE)
target_include_directories(crow INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/include)
