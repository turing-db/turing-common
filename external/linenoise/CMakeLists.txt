
add_library(linenoise_s STATIC linenoise.c)
target_include_directories(linenoise_s PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})

