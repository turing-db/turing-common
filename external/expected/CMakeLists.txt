project(expected VERSION 0.8.0)

add_library(expected INTERFACE)
add_library(expected::expected ALIAS expected)
target_include_directories(expected INTERFACE include)
